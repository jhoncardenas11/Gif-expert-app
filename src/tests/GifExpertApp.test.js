import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas en gifExpertApp.js', () => {

    test('Debe hacer match con el snapshot', () => {
        const wrapper = shallow( <GifExpertApp /> );
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe de mostrar una lista de categorias', () => {
        const categories = ['cowboy bebop', 'evangelion'];

        const wrapper = shallow( <GifExpertApp defaultCategories={categories}/> );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })

});