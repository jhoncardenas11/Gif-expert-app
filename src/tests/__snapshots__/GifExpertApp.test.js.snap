// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en gifExpertApp.js Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="cowboy bebop"
      key="cowboy bebop"
    />
    <GifGrid
      category="evangelion"
      key="evangelion"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en gifExpertApp.js Debe hacer match con el snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
